yarn-project/foundation/src/fifo/memory_fifo.ts

```diff
@@ -91,9 +91,9 @@ export class MemoryFifo<T> {
 
   /**
    * Process items from the queue using a provided handler function.
-   * The function iterates over items in the queue, invoking the handler for each item until the queue is empty or flushing.
+   * The function iterates over items in the queue, invoking the handler for each item until the queue is empty and flushing.
```

Comment:
See the `get` function above. the `process` loop here only breaks when we get `null`, which only happens `if (this.items.length === 0 && this.flushing)`